sklearn/model_selection/_search.py

```diff
@@ -587,6 +587,13 @@ def fit(self, X, y=None, groups=None, **fit_params):
                               'the "fit" method.', RuntimeWarning)
             else:
                 fit_params = self.fit_params
+        if self.return_train_score == "warn":
+            warnings.warn("Computing training scores may affect "
+                          "performance significantly. This is the reason "
```

Comment:
"may affect performance" is ambiguous: it could be interpreted as affecting the predictive accuracy. I would suggest: "can slow do the grid search significantly" or "may be computationally expensive".